src/platforms/salesforce/schemas.ts

```diff
@@ -888,6 +888,64 @@ export const salesforceListViewResult = z.object({
   size: z.number(),
 });
 
+// -
+// Custom Fields
+// -
+const salesforceCustomFieldValueSetDefinition = z.object({
+  sorted: z.boolean().optional(),
+  value: z.array(
+    z.object({
+      fullName: z.string().optional(),
+      default: z.boolean().optional(),
+      label: z.string(),
+      valueName: z.string().optional(),
+    }),
+  ),
+});
+
+const salesforceCustomFieldMetadata = z.object({
+  label: z.string(),
+  type: z.string(),
+  writeRequiresMasterRead: z.boolean().optional(),
+  required: z.boolean(),
+  valueSet: z
+    .object({
+      valueSetDefinition: salesforceCustomFieldValueSetDefinition,
+    })
+    .optional(),
+  visibleLines: z.number().optional(),
+  length: z.number().optional(),
+  precision: z.number().optional(),
+  scale: z.number().optional(),
+});
+
+const salesforceCustomField = z.object({
+  FullName: z.string().optional(),
+  Metadata: salesforceCustomFieldMetadata,
+});
+
+const salesforceCustomFieldCreate = z.object({
+  CustomField: z.object({
+    FullName: z.string().optional(),
+    Metadata: salesforceCustomFieldMetadata.extend({
+      type: z
+        .string()
+        .transform((val) => val as Capitalize<SalesforceFieldType>),
```

Comment:
this is an input, right? shouldn't it take an enum?

Reply:
The z.enum() function only takes in a Readonly<[string, ...string[]> so it looked a bit uglier to convert `SALESFORCE_FIELD_TYPES.map(capitalize)` into that